### ISSUE TITLE
Positional arguments and "strange" errors.

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#69]
By reading your reply to the issue #68 I've suddenly realized, that it can be that the issue of the "strange" outputs we are experiencing is due to the fact we are suppressing the flags, thus we are forced to put the values of those flags into certain positions.

So I've tested `transaction list -x --_value1 mfbn3FPxjUXKGNafLhPa6q5SxATPCUgj8w -f 396100 -e 396205` for instance and got the same output of `transaction list mfbn3FPxjUXKGNafLhPa6q5SxATPCUgj8w -x -f 396100 -e 396205`.
Moreover the `transaction list -x -f 396100 -e 396205 mfbn3FPxjUXKGNafLhPa6q5SxATPCUgj8w` command has also provided me the same output.

While this is probably a Pulcinella's secret for you, I'm thinking whether it would make sense to let our users know that they have the possibility to choose whatever position they like for the positional arguments provided they'd be using the right flags before.
That of course would provide us a new challenge on how to call those flags. And probably the idea goes into the opposite direction you were working so far. So I'm not insisting on this.

**Comment by d5000** [comment]
You nailed it with your last sentence ;)

I spent lots of time thinking about this issue. My conclusion is that if a command *can* be used as positional argument (what you describe as "suppress the flag"), the user *should* use it that way, because otherwise one would need to explain why he can use it in a positional way or not, what would be the difference ... etc.. It's not worth it and would make even the manual more complicated and confusing.

This is the reason why I gave the arguments such strange names like `_value1`, users should not even be tempted to think that they should put the flag name (keyword) in some place (before I had experimented with keywords like `address_or_deck` but I thought this would be even more confusing for the user). It's all for simplicity.

Actually these optional positional arguments make the whole CLI more complicated. But if we removed them, we wouldn't have such nice commands like `address set LABEL ADDRESS` or `transaction list ADDRESS`.

> Moreover the transaction list -x -f 396100 -e 396205 mfbn3FPxjUXKGNafLhPa6q5SxATPCUgj8w command has also provided me the same output.

This one surprised me, but it works ... i.e. the address is not ignored but read as the first positional argument. So Fire actually processes keyword arguments completely independent from positional arguments. I would however not like to explain this to the users, they should use the command in the order it's given in the manual. It would only lead to more confusion.

I think we can close, I think the current state of things with the CLI is reasonably good, because if the user follows the manual/docstring the commands will work correctly. As I wrote in the other issue, once the commands are "stable" enough, I'll add a comment to the manual that the order and number of positional arguments should be really respected because otherwise it could lead to errors.
